craft_parts/parts.py

```diff
@@ -64,36 +81,39 @@ class PartSpec(BaseModel):
     override_stage: Optional[str] = None
     override_prime: Optional[str] = None
     permissions: List[Permissions] = []
-
-    class Config:
-        """Pydantic model configuration."""
-
-        validate_assignment = True
-        extra = "forbid"
-        allow_mutation = False
-        alias_generator = lambda s: s.replace("_", "-")  # noqa: E731
+    model_config = ConfigDict(
+        validate_assignment=True,
+        extra="forbid",
+        frozen=True,
+        strict=True,
```

Comment:
specifically, this would break because `source-tag` is a float and not a string right? I think this is "too breaking" of a change: it would break all snaps, rocks, charms etc that use things that look like numbers.

This is also mostly unrelated to migrating to v2: if possible, we should stick to the minimum required to do the update with as few changes as possible.

Reply:
I replied to Tiago's comment below before seeing this comment.
Now I get a better view of this issue, I agree strict=True should be "strictly" reverted.